components/core/src/VariableDictionaryWriter.cpp

```diff
@@ -70,16 +69,17 @@ bool VariableDictionaryWriter::add_occurrence (const string& value, variable_dic
 
         // Insert the ID obtained from the database into the dictionary
         auto* entry = new VariableDictionaryEntry(value, id);
-        m_value_to_entry[value] = entry;
-        m_id_to_entry[id] = entry;
-
-        // Mark ID as dirty
-        m_uncommitted_entries.emplace_back(entry);
+        m_value_to_id[value] = id;
 
         new_entry = true;
 
         // TODO: This doesn't account for the segment index that's constantly updated
+        // Nor does it account that compression doesn't store the whole entry in the memory
```

Comment:
Let's leave this comment out for now since I think it will confuse readers even more, lol. The issue is that we're using this size to determine when to split an archive (`m_target_data_size_of_dictionaries`), which to me sounds like it should refer to the on-disk uncompressed size. However, the current comments (before your change) make it sound like it really refers to the in-memory size. I'll fix the discrepancy in another commit.